driver/utils/unicode_conv.h

```diff
@@ -11,6 +11,20 @@
 
 #include <cstring>
 
+class UnicodeConversionContext {
```

Comment:
I would personally prefer a clean interface, like:
```
class UnicodeConverter {
    void convertFromUtf8ToUtf16(const char * in, char16_t * out);
// ... and similar
};
```

Then you can change the implementation without re-doing the rest of the system to use ICU (or another library).

Reply:
I prefer free functions over member functionsin such casesw, when their code is generic and can be instantiated given some conformant templated "context" argument.